Sources/Graphiti/Subscription/SubscribeField.swift

```diff
@@ -0,0 +1,409 @@
+import GraphQL
+import Runtime
+import RxSwift
+
+// Subscription resolver must return an Observer<Any>, not a specific type, due to lack of support for covariance generics in Swift
+
+public class SubscriptionField<ObjectType, Context, FieldType, Arguments : Decodable> : FieldComponent<ObjectType, Context> {
+    let name: String
+    let arguments: [ArgumentComponent<Arguments>]
+    let resolve: GraphQLFieldResolve
+    let subscribe: GraphQLFieldResolve
+    
+    override func field(typeProvider: TypeProvider) throws -> (String, GraphQLField) {
+        let field = GraphQLField(
+            type: try typeProvider.getOutputType(from: FieldType.self, field: name),
+            description: description,
+            deprecationReason: deprecationReason,
+            args: try arguments(typeProvider: typeProvider),
+            resolve: resolve,
+            subscribe: subscribe
+        )
+        
+        return (name, field)
+    }
+    
+    func arguments(typeProvider: TypeProvider) throws -> GraphQLArgumentConfigMap {
+        var map: GraphQLArgumentConfigMap = [:]
+        
+        for argument in arguments {
+            let (name, argument) = try argument.argument(typeProvider: typeProvider)
+            map[name] = argument
+        }
+        
+        return map
+    }
+    
+    init(
+        name: String,
+        arguments: [ArgumentComponent<Arguments>],
+        resolve: @escaping GraphQLFieldResolve,
+        subscribe: @escaping GraphQLFieldResolve
+    ) {
+        self.name = name
+        self.arguments = arguments
+        self.resolve = resolve
+        self.subscribe = subscribe
+    }
+
+    convenience init<SourceEventType, ResolveType>(
+        name: String,
+        arguments: [ArgumentComponent<Arguments>],
+        asyncResolve: @escaping AsyncResolve<SourceEventType, Context, Arguments, ResolveType>,
+        asyncSubscribe: @escaping AsyncResolve<ObjectType, Context, Arguments, Observable<Any>>
+    ) {
+        let resolve: GraphQLFieldResolve = { source, arguments, context, eventLoopGroup, _ in
+            guard let s = source as? SourceEventType else {
```

Comment:
In swift you can actually re-scope an optional using the same variable name, e.g
```suggestion
            guard let source = source as? SourceEventType else {
```

Reply:
Oh, that's awesome! Thanks for the tip.